File: backends/java/zappy-application/src/main/java/com/zappymart/application/ports/PromotionRepository.java
package com.zappymart.application.ports;

import com.zappymart.domain.promotions.Promotion;
import com.zappymart.domain.promotions.PromotionCode;

import java.util.Optional;

public interface PromotionRepository {

    Optional<Promotion> byCode(PromotionCode code);

    void save(Promotion promotion);
}

File: backends/kotlin/zappy-application/src/main/kotlin/nl/zappymart/application/ports/PromotionRepository.kt
package nl.zappymart.application.ports

import nl.zappymart.domain.promotions.Promotion
import nl.zappymart.domain.shared.PromotionCode

interface PromotionRepository {

    fun findByCode(code: PromotionCode): Promotion?

    fun save(promotion: Promotion)
}
